src/app/activity/pages/activity-statistics/activity-statistics.component.ts

```diff
@@ -1,12 +1,322 @@
+  };
+  currentYear = null;
+  colors = ['#d51c54', '#f8b231', '#00a19a'];
 
-  ngOnInit(): void {}
+  filters = new FormGroup({
+    year: new FormControl(),
+  });
+
+  subscriptions: Subscription[] = [];
+  constructor(
+    private myStatsGQL: MyActivityStatsGQL,
+    private loadingSpinnerService: LoadingSpinnerService,
+    private authService: AuthService,
+    private GradingSystemsService: GradingSystemsService
+
+  ) {}
+
+  @Output() errorEvent = new EventEmitter<DataError>();
+
+  ngOnInit(): void {
+    const filtersSub = this.filters.valueChanges.subscribe((values) => {
+      this.currentYear = values.year;
+      this.querySuccess(values.year);
+    });
+    this.subscriptions.push(filtersSub);
+
+    this.subscription = this.authService.currentUser
+      .pipe(
+        switchMap((user) => {
+          this.loadingSpinnerService.pushLoader();
+          return this.myStatsGQL.fetch({
+            input: {}
+          });
+        })
+      )
+      .subscribe({
+        next: (result) => {
+          this.loading = false;
+          this.myStats = <StatsActivities[]>result.data.myActivityStatistics;
+          this.querySuccess(null);
+          this.parseByYear();
+
+        },
+        error: (error) => {
+          this.loadingSpinnerService.popLoader();
+          this.queryError();
+        },
+      });
+
+  }
+
+  //parses result for graph by year
+  parseByYear(): void {
+    this.activityYears.forEach(year => {
+      if(year.value) {
+          var filterByYear = this.myStats.filter(x=> x.year == year.value);
+          filterByYear.forEach((elementByYear, index) => {
+            if(elementByYear.ascent_type=='redpoint') year.nrRoutesRP += elementByYear.nr_routes;
+            if(elementByYear.ascent_type=='flash') year.nrRoutesF += elementByYear.nr_routes;
+            if(elementByYear.ascent_type=='onsight') year.nrRoutesOS += elementByYear.nr_routes;
+          });
+      }
+    });
+
+    var sorted = this.activityYears.filter(year => year.value).sort((a, b) => a.value - b.value);
+    this.buildOptionsLine(sorted);
+  }
+
+  async querySuccess(year) {
+    var data1 = [];
+    var data2 = [];
+    var data3 = [];       
+    let xAxisLabels = [];  
+
+    this.myStats.forEach(async (element) => {
+      if(year && element.year!==year) return;
+      if( this.activityYears.findIndex((el) => el.value === element.year)===-1)  {
+        this.activityYears.push({
+          value: element.year,
+          label: element.year.toString(),
+          nrRoutesRP: 0,
+          nrRoutesF: 0,
+          nrRoutesOS: 0
+        }); 
+      }
+
+      this.activityYears = this.activityYears.sort((a, b) => {
+        if(a.value == null || b.value == null) -1;
+        else return b.value - a.value
+      })
+
+      if(!year) this.currentYear = this.activityYears[0].label;
+
+      try {
+        var grade = await this.GradingSystemsService.diffToGrade(
+          element.difficulty,
+          'french',
+          true
```

Comment:
This should be false, intermediate grades are (currently) only displayed for legacy diff votes.

Reply:
Changed legacy param to false and changed colors to match color scheme.